extensions/ql-vscode/src/remote-queries/gh-actions-api-client.ts

```diff
@@ -255,6 +254,44 @@ function tryGetArtifactIDfromName(
   return artifact?.id;
 }
 
+/**
+ * Wait for an artifact to be available in a workflow run.
+ * @param credentials Credentials for authenticating to the GitHub API.
+ * @param owner
+ * @param repo
+ * @param workflowRunId The ID of the workflow run to get the artifact for.
+ * @param artifactName The artifact name, as a string.
+ * @param maxAttempts The maximum number of attempts to download the artifact.
+ * @returns An array containing the full list of artifacts and the ID of the artifact with the given name.
+ */
+async function waitForArtifact(
+  credentials: Credentials,
+  owner: string,
+  repo: string,
+  workflowRunId: number,
+  artifactName: string,
+  maxAttempts = 10,
+  intervalBetweenAttempts = 1000
```

Comment:
Better to use exponential backoff here? Eg- interval of 1s, 2s, 4s, 8s, ....?

Reply:
I would expect Actions to make an artifact available reasonably soon, in the order of seconds rather than minutes. I don't think exponential back-off would have been necessary for this.